### PR TITLE
DAOS-3943 Enable selftest to run with test_group

### PR DESCRIPTION
**Pull request by sylviachanoiyee**
Enable selftest to run with test_group

### CODE REVIEW

File: run_cart.sh [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            done
+
+            let DAOS_SERVERS=126
+            prepare
+            run_selftest
+            ;;
+    esac
+done
+
+mv stderr* stdout* testLogs Log/$SLURM_JOB_ID
```
Comment (ravalsam) [inline run_cart.sh]: This should be specific to the current job stdout.o%j and stderr.e%j .

File: run_cart.sh [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        if [ $ret != 0 ]; then
+                echo TEST FAILED!!! ret = $ret
+        else
+                echo TEST PASSED
+        fi
+        echo ""
+}
+
+#Create server/client hostfile.
+prepare(){
```
Comment (ravalsam) [inline run_cart.sh]: May be better this to move to common script instead of duplicate in each run?

File: run_cart.sh [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            ;;
+        self_test)
+            let DAOS_SERVER=2
+            until [  $DAOS_SERVERS -gt 65 ]; do
+                prepare
+                run_selftest
+                let DAOS_SERVERS=$(( DAOS_SERVERS*2 ))
+            done
+
+            let DAOS_SERVERS=126
```
Comment (ravalsam) [inline run_cart.sh]: Better to use another variable on top of the script MAX_SERVERS and assign it to DAOS_SERVER?

File: run_cart.sh (hunk 2 above)
Comment (sylviachanoiyee) [reply run_cart.sh]: Sure.

File: run_cart.sh (hunk 3 above)
Comment (sylviachanoiyee) [reply run_cart.sh]: I think using the same var should be fine. However, i did rename the var in run_cart to better reflect the generic usage of server and client.

File: run_cart.sh (hunk 1 above)
Comment (sylviachanoiyee) [reply run_cart.sh]: sure

File: run_cart.sh [hunk 4]
```diff
@@ -101,47 +104,43 @@ prepare(){
...
+            let $N_SERVER=2
+            until [  $N_SERVERS -gt 65 ]; do
                 prepare
                 run_selftest
-                let DAOS_SERVERS=$(( DAOS_SERVERS*2 ))
+                let $N_SERVERS=$(( $N_SERVERS*2 ))
             done
 
-            let DAOS_SERVERS=126
+            let $N_SERVERS=126
```
Comment (ravalsam) [inline run_cart.sh]: My previous comment is suppose to be related to number 126. Are we always expected to run with 126? If we have skx-large what that 126 number will be ? Is that suppose to be N_SERVER -1 ? or am I misunderstanding something related to 126 server?

Comment (sylviachanoiyee) [reply run_cart.sh]: ah... copy and paste from my run for executing on 128 skx-normal cluster.
let me fix this.

File: run_cart.sh [hunk 5]
```diff
@@ -7,17 +7,17 @@
 #SBATCH -o stdout.o%j           # Name of stdout output file
 #SBATCH -e stderr.e%j           # Name of stderr error file
 #SBATCH -A STAR-Intel           # Project Name
-#SBATCH -p skx-dev              # Queue (partition) name
-#SBATCH -N 4                    # Total # of nodes
-#SBATCH -n 40                   # Total # of mpi tasks (48 x  Total # of nodes)
+#SBATCH -p skx-normal           # Queue (partition) name
+#SBATCH -N 10                   # Total # of nodes
+#SBATCH -n 100                  # Total # of mpi tasks (48 x  Total # of nodes)
```
Comment (saurabhtandan) [inline run_cart.sh]: Minor: Seems like you are only requesting 10 tasks per node but in comment it says (48 x num of nodes).
Just wanted to bring it in your attention.

**Pull request merged**